pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/RegexpLikePredicateEvaluatorFactory.java

```diff
@@ -121,15 +133,22 @@ private static final class ScanBasedRegexpLikePredicateEvaluator extends BaseDic
     // Reuse matcher to avoid excessive allocation. This is safe to do because the evaluator is always used
     // within the scope of a single thread.
     final Matcher _matcher;
+    Int2BooleanMap _dictIdMap;
```

Comment:
(minor) This can be `final`.

Let's also add some comments explaining the algorithm